examples/siro_sandbox/controllers/controller_helper.py

```diff
@@ -97,23 +98,39 @@ def __init__(
                     gui_input=gui_input,
                 )
             self.controllers.append(gui_agent_controller)
-
             if is_multi_agent:
-                self.controllers.append(
-                    SingleAgentBaselinesController(
-                        0 if self._gui_controlled_agent_index == 1 else 1,
-                        is_multi_agent,
-                        config,
-                        self._gym_habitat_env,
+                if args.app_state == "fetch":
```

Comment:
My concern is that we'll forget about this app-state-specific logic here in this file, and it'll get confusing to add new app states.

Let me request this refactor: let's avoid giving `ControllerHelper` the whole args object and instead just pass only what it needs from there: `app_state`, `gui_controlled_agent_index`, and `walk_pose_path`. At least then, the calling code to construct the ControllerHelper in SandboxDriver will be more clear about what ControllerHelper is doing.